lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -432,9 +425,24 @@ void SPIRVRegularizeLLVMBase::expandSYCLTypeUsing(Module *M) {
 
 Value *SPIRVRegularizeLLVMBase::extendBitInstBoolArg(Instruction *II) {
   IRBuilder<> Builder(II);
-  auto *NewBase = Builder.CreateZExt(II->getOperand(0), Builder.getInt32Ty());
-  auto *ShiftInt32 = cast<User>(II->getOperand(1))->getOperand(0);
-  return Builder.CreateLShr(NewBase, ShiftInt32);
+  auto *ArgTy = II->getOperand(0)->getType();
+  Type *NewArgType = nullptr;
+  if (ArgTy->isIntegerTy()) {
+    NewArgType = Builder.getInt32Ty();
+  } else {
```

Comment:
I see that the function is called only for integers and vectors of integers now, but it's not guaranteed later.
I suggest we check for the vector type here too, e.g. `else if (ArgTy->isIntVectorTy())`. This will require dealing with the `else` case (e.g. warning + `return II`, or something else like asserts if you find it right). It would be helpful if the function would be misused in the future.

Reply:
Thanks, [2a8c9e3](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1450/commits/2a8c9e3ed30da49314b7cd464c45597f646674d6)